docs(docs-infra): Updates page title strategy to sync meta tags

SkyZeroZx opened the pull request:
Ensures that page title changes are reflected in relevant meta tags for improved SEO and sharing support

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
 
<img width="356" height="327" alt="image" src="https://github.com/user-attachments/assets/350a1101-e595-4154-b43b-7ba7d4114474" />


Issue Number: N/A


## What is the new behavior?
<img width="359" height="346" alt="image" src="https://github.com/user-attachments/assets/07e2f12c-40e3-4834-ae19-5ff584a7d1d5" />


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche reviewed (approved):
Great, thanks for this ! This is something I wanted to improve.

JeanMeche commented:
Yeah, let's remove that meta tag from the `index.html`. I couldn't find any source that confirms that this tag is valid.

SkyZeroZx committed:
docs(docs-infra): Updates page title strategy to sync meta tags

Ensures that page title changes are reflected in relevant meta tags for improved SEO and sharing support
